### ISSUE TITLE
Deprecation message symfony 6.3

### ISSUE THREAD

**Issue by phpremco** [doctrine-extensions/DoctrineExtensions#2664]
### Environment
local ddev symfony 6.3 / linux

#### Package
gedmo/doctrine-extensions

<details><summary>show</summary>
<p>

```
$ composer show --latest gedmo/doctrine-extensions
Info from https://repo.packagist.org: #StandWithUkraine
name     : gedmo/doctrine-extensions
descrip. : Doctrine behavioral extensions
keywords : Blameable, behaviors, doctrine, extensions, gedmo, loggable, nest
edset, odm, orm, sluggable, sortable, timestampable, translatable, tree, uploadable
versions : * 3.13.x-dev, * dev-main
latest   : dev-main
type     : library
license  : MIT License (MIT) (OSI approved) https://spdx.org/licenses/MIT.html#licenseText
homepage : http://gediminasm.org/
source   : [git] https://github.com/doctrine-extensions/DoctrineExtensions.git ef277c95ad0660a69313f9cb6b53497e5d2e8f4c
dist     : [zip] https://api.github.com/repos/doctrine-extensions/DoctrineEx
tensions/zipball/ef277c95ad0660a69313f9cb6b53497e5d2e8f4c ef277c95ad0660a69313f9cb6b53497e5d2e8f4c
path     : /var/www/html/vendor/gedmo/doctrine-extensions
names    : gedmo/doctrine-extensions

support
email : gediminas.morkevicius@gmail.com
issues : https://github.com/doctrine-extensions/DoctrineExtensions/issues   
source : https://github.com/doctrine-extensions/DoctrineExtensions/tree/main
wiki : https://github.com/Atlantic18/DoctrineExtensions/tree/main/doc       

autoload
psr-4
Gedmo\ => src/

requires
behat/transliterator ^1.2
doctrine/annotations ^1.13 || ^2.0
doctrine/collections ^1.2 || ^2.0
doctrine/common ^2.13 || ^3.0
doctrine/event-manager ^1.2 || ^2.0
doctrine/persistence ^2.2 || ^3.0
php ^7.2 || ^8.0
psr/cache ^1 || ^2 || ^3
symfony/cache ^4.4 || ^5.3 || ^6.0
symfony/deprecation-contracts ^2.1 || ^3.0

requires (dev)
doctrine/cache ^1.11 || ^2.0
doctrine/dbal ^2.13.1 || ^3.2
doctrine/doctrine-bundle ^2.3
doctrine/mongodb-odm ^2.3
doctrine/orm ^2.10.2
friendsofphp/php-cs-fixer ^3.4.0 <3.10
nesbot/carbon ^2.55
phpstan/phpstan ^1.10.2
phpstan/phpstan-doctrine ^1.0
phpstan/phpstan-phpunit ^1.0
phpunit/phpunit ^8.5 || ^9.5
rector/rector ^0.15.20
symfony/console ^4.4 || ^5.3 || ^6.0
symfony/phpunit-bridge ^6.0
symfony/yaml ^4.4 || ^5.3 || ^6.0

suggests
doctrine/mongodb-odm to use the extensions with the MongoDB ODM
doctrine/orm to use the extensions with the ORM

conflicts
doctrine/dbal <2.13.1 || ^3.0 <3.2
doctrine/mongodb-odm <2.3
doctrine/orm <2.10.2
sebastian/comparator <2.0

```

</p>
</details>

#### Doctrine packages

<details><summary>show</summary>
<p>

```
$ composer show --latest 'doctrine/*'
Color legend:
- patch or minor release available - update recommended
- major release available - update possible
- up to date version

Direct dependencies required in composer.json:
doctrine/doctrine-migrations-bundle 3.2.4  3.2.4  Symfony DoctrineMigrationsBundle
doctrine/orm                        2.16.0 2.16.0 Object-Relational-Mapper for PHP

Transitive dependencies not required in composer.json:
doctrine/annotations                2.0.1  2.0.1  Docblock Annotations Parser
doctrine/cache                      2.2.0  2.2.0  PHP Doctrine Cache library is a popular cache impleme...
doctrine/collections                2.1.3  2.1.3  PHP Doctrine Collections library that adds additional...
doctrine/common                     3.4.3  3.4.3  PHP Doctrine Common project is a library that provide...
doctrine/dbal                       3.6.5  3.6.5  Powerful PHP database abstraction layer (DBAL) with m...
doctrine/deprecations               v1.1.1 v1.1.1 A small layer on top of trigger_error(E_USER_DEPRECAT...
doctrine/doctrine-bundle            2.10.1 2.10.1 Symfony DoctrineBundle    
doctrine/event-manager              2.0.0  2.0.0  The Doctrine Event Manager is a simple PHP event syst...
doctrine/inflector                  2.0.8  2.0.8  PHP Doctrine Inflector is a small library that can pe...
doctrine/instantiator               2.0.0  2.0.0  A small, lightweight utility to instantiate objects i...
doctrine/lexer                      2.1.0  3.0.0  PHP Doctrine Lexer parser library that can be used in...
doctrine/migrations                 3.6.0  3.6.0  PHP Doctrine Migrations project offer additional func...
doctrine/persistence                3.2.0  3.2.0  The Doctrine Persistence project is a set of shared i...
doctrine/sql-formatter              1.1.3  1.1.3  a PHP SQL highlighting library

```

</p>
</details>

#### PHP version

```
$ php -v
PHP 8.2.3 (cli) (built: Feb 14 2023 16:53:07) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.3, Copyright (c) Zend Technologies
    with Zend OPcache v8.2.3, Copyright (c), by Zend Technologies

```

## Subject
ORM.php:98 calls getEntityManager(), this has been changed to getObjectManager(), it is causing deprecation messages on entity persist

## Steps to reproduce
I ran an entity save in a command, this message is thrown. The entity uses a timestampable on 2 properties.

## Expected results
No deprecation message.

## Actual results
2023-08-06T20:58:01+02:00 [info] User Deprecated: Method Doctrine\ORM\Event\OnFlushEventArgs::getEntityMan
ager() is deprecated and will be removed in Doctrine ORM 3.0. Use getObjectManager() instead. (OnFlushEventArgs.php:29 called by ORM.php:98, https://github.com/doctrine/orm/issues/9875, package doctrine/orm)

**Comment by ovgray** [comment]
Calling getEntityManager on most (but not all) eventArgs types is deprecated in doctrine/orm since ver 2.13.

Maybe replace line 98 of Gedmo\Mapping\Event\Adapter\ORM with 

`return method_exists($this->args, 'getObjectManager') ? $this->args->getObjectManager() : $this->args->getEntityManager();`

**Comment by phpremco** [comment]
Thanks, yes that works like a charm. I had to do the same for getEntity.
The following has fixed my issue:

    /**
     * @return EntityManagerInterface
     */
    public function getObjectManager()
    {
        if (null !== $this->em) {
            return $this->em;
        }

        if (null === $this->args) {
            throw new \LogicException(sprintf('Event args must be set before calling "%s()".', __METHOD__));
        }

        return method_exists($this->args, 'getObjectManager') ? $this->args->getObjectManager() : $this->args->getEntityManager();
    }

    public function getObject(): object
    {
        if (null === $this->args) {
            throw new \LogicException(sprintf('Event args must be set before calling "%s()".', __METHOD__));
        }
        
        return method_exists($this->args, 'getObject') ? $this->args->getObject() : $this->args->getEntity();
    }

**Comment by misaert** [comment]
Hi,

It seems fix by #2639, right?

**Comment by franmomu** [comment]
> Hi,
> 
> It seems fix by #2639, right?

yes, you're right, thanks for pointing it out
